### PR TITLE
add auto-backport functionality for reporting

### LINKED ISSUE #237: Add auto backport functionality to backport PRs for releases
Coming from meta issue [opensearch-project/opensearch-plugins#108](https://github.com/opensearch-project/opensearch-plugins/issues/108):

- [ ] Add auto-backport PR creation Github action to make sure all the PRs are backported for a particular release. 
- [ ] Add auto-delete for merged auto-backport PRs to clean up the branches created during auto-backport.
- [ ] Add documentation for the same.

Related documentation: https://github.com/opensearch-project/opensearch-plugins/blob/main/BACKPORT.md

### PR DESCRIPTION
**Pull request by davidcui1225**
Signed-off-by: David Cui <davidcui@amazon.com>

### Description
Add auto-backport functionality to reporting.

### Issues Resolved
#237 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by davidcui1225** [comment]
Link checker is failing even though I can access the links on browser, investigating

File: dashboards-reports/DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -44,3 +44,9 @@ Example output: `./build/reports-dashboards-0.0.1.zip`
 - `yarn test:jest`
 
   Runs the plugin tests.
+
+### Backports
+
+The Github workflow in [`backport.yml`](../.github/workflows/backport.yml) creates backport PRs automatically when the
+original PR with an appropriate label `backport <backport-branch-name>` is merged to main with the backport workflow
+run successfully on the PR. For example, if a PR on main needs to be backported to `1.x` branch, add a label
```
Comment (joshuali925) [inline dashboards-reports/DEVELOPER_GUIDE.md:52]: i don't think we use line-breaks to format in markdown files? And should this be 
```suggestion
run successfully on the PR. For example, if a PR on main needs to be backported to `1.x` branch, add a label `backport 1.x`.
```

**Review by joshuali925 — DISMISSED** [review]

File: dashboards-reports/DEVELOPER_GUIDE.md (hunk 1 above)
Comment (davidcui1225) [reply dashboards-reports/DEVELOPER_GUIDE.md:52]: Yes good catch, I missed the whole message in my paste. Also any idea why the link checker might be failing? I can access all of the URLs in my browser as they are listed in the failed action's raw logs

Comment (joshuali925) [reply dashboards-reports/DEVELOPER_GUIDE.md]: hmm not sure, might be network issue, retry it later?

**Review by zhongnansu — DISMISSED** [review]

**Review by joshuali925 — APPROVED** [review]

**Review by eugenesk24 — DISMISSED** [review]
🚢

**Comment by ps48** [comment]
You might want to fix link to `backport.yml` file in developer guide.

**Comment by davidcui1225** [comment]
> You might want to fix link to `backport.yml` file in developer guide.

Good catch

**Review by joshuali925 — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

**Pull request merged**